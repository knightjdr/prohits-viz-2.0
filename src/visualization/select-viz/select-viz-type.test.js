import React from 'react';
import { shallow } from 'enzyme';

import SelectVizType from './select-viz-type';

const handleFile = jest.fn();

describe('SelectVizType', () => {
  test('Renders with upload option when vizType is null', () => {
    const wrapper = shallow(
      <SelectVizType handleFile={handleFile} />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('HeatmapContainer').length).toBe(0);
    expect(wrapper.find('ScatterContainer').length).toBe(0);
    expect(wrapper.find('Selection').length).toBe(1);
  });

  test('Renders with HeatmapContainer', () => {
    const wrapper = shallow(
      <SelectVizType
        handleFile={handleFile}
        vizType="heatmap"
      />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('HeatmapContainer').length).toBe(1);
    expect(wrapper.find('ScatterContainer').length).toBe(0);
    expect(wrapper.find('Selection').length).toBe(0);
  });

  test('Renders with ScatterContainer', () => {
    const wrapper = shallow(
      <SelectVizType
        handleFile={handleFile}
        vizType="scatter"
      />,
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('HeatmapContainer').length).toBe(0);
    expect(wrapper.find('ScatterContainer').length).toBe(1);
    expect(wrapper.find('Selection').length).toBe(0);
  });
});
