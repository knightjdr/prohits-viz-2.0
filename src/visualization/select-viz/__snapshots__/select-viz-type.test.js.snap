// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectVizType Renders with HeatmapContainer 1`] = `<HeatmapContainer />`;

exports[`SelectVizType Renders with ScatterContainer 1`] = `<ScatterContainer />`;

exports[`SelectVizType Renders with upload option when vizType is null 1`] = `
<Selection
  err={null}
  handleFile={[Function]}
/>
`;
